### PR TITLE
add [sig-apps] prefix to workload e2e tests

### PR DESCRIPTION
**Pull request by crimsonfaith91**
**What this PR does / why we need it**: This PR adds [sig-apps] prefix to workload e2e tests in accord to requirements of adding a SIG dashboard to testgrid. Refer PR #48781 for guidelines.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: test/e2e/workload/statefulset.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ var _ = framework.KubeDescribe("StatefulSet", func() {
 		ns = f.Namespace.Name
 	})
 
-	framework.KubeDescribe("Basic StatefulSet functionality [StatefulSetBasic]", func() {
```
Comment (foxish) [inline test/e2e/workload/statefulset.go]: This I think is nested within the outer `SIGDescribe` and would add the SIG label twice.

File: test/e2e/workload/statefulset.go [hunk 2]
```diff
@@ -864,7 +864,7 @@ var _ = framework.KubeDescribe("StatefulSet", func() {
 		})
 	})
 
-	framework.KubeDescribe("Deploy clustered applications [Feature:StatefulSet] [Slow]", func() {
```
Comment (foxish) [inline test/e2e/workload/statefulset.go]: Same as above

File: test/e2e/workload/statefulset.go (hunk 1 above)
Comment (crimsonfaith91) [inline test/e2e/workload/statefulset.go]: noted! good catch!

**Comment by spiffxp** [comment]
ref https://github.com/kubernetes/kubernetes/issues/49161

**Pull request merged**